db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ResultColumn.java

```diff
@@ -1510,6 +1511,13 @@ void rejectParameter() throws StandardException
             throw StandardException.newException(SQLState.LANG_PARAM_IN_SELECT_LIST);
     }
 
+    void setUnknownParameterType(DataTypeDescriptor type) throws StandardException
+    {
+        if (expression != null && expression.isParameterNode()) {
```

Comment:
This looks a bit strange to me, should this be an assertion instead, i.e. isn't a programming error to set a parameter type to a non-parameter column?

Reply:
This function is modelled after rejectParameterColumn. I could rename it to setPotentialUnknownParameterType if that makes more sense. The idea being that we want to set the type iff it's a parameterNode